### PR TITLE
Add encoding param for paths

### PR DESCRIPTION
**Pull request by scott-8**

### CODE REVIEW

File: README.rst
Comment (shapiromatron) [inline README.rst:37]: change parameter name to `encoding`? to be consistent with python open()  - https://docs.python.org/3/howto/unicode.html#reading-and-writing-unicode-data

File: rispy/parser.py
Comment (shapiromatron) [inline rispy/parser.py:190]: does this need to change after #36 is merged in? you'll need to pop encoding off the kw stack right?

File: tests/test_parser.py
Comment (shapiromatron) [inline tests/test_parser.py:307]: use `pytest.raises` here https://docs.pytest.org/en/reorganize-docs/new-docs/user/pytest_raises.html

File: tests/test_parser.py
Comment (shapiromatron) [inline tests/test_parser.py:304]: i think you can remove this 304-305

**Review by shapiromatron — CHANGES_REQUESTED** [review]
looks good - a few changes, but maybe we should wait until #36 is merged in?

File: rispy/parser.py
Comment (scott-8) [reply rispy/parser.py:190]: Yeah I was planning on waiting until #36 is in before merging this

File: README.rst
Comment (scott-8) [reply README.rst:37]: I thought about it earlier, the reason I chose `path_encoding` was because I wanted to specify that it was only useful when you supplied a path. it's probably still better to leave it as `encoding`, though, and just document it

File: tests/test_parser.py
Comment (scott-8) [reply tests/test_parser.py:304]: The reason it's there is to compare the different methods of opening. There's a bug with read_text where it reads certain characters incorrectly, so this tests that. see: https://stackoverflow.com/questions/49428723

File: tests/test_parser.py
Comment (scott-8) [reply tests/test_parser.py:307]: thanks, didn't know this was a thing

File: tests/test_parser.py
Comment (scott-8) [reply tests/test_parser.py:304]: although I guess I'm not sure this is something that's worth testing here

File: README.rst
Comment (J535D165) [inline README.rst:36]:
```
p = Path('tests', 'data', 'example_utf_chars.ris')
```

File: README.rst
Comment (J535D165) [inline README.rst:31]: Unused import?

File: README.rst
Comment (J535D165) [inline README.rst:33]: Unused import?

File: rispy/parser.py
Comment (J535D165) [inline rispy/parser.py:190]: I don't think this is the best way to implement this. I think it should be the second position (the other arguments are options of the parser) and not with a variable-length argument. 

```
def load(    
    file: Union[TextIO, Path],
    encoding: str,
    mapping: Optional[Dict] = None,
    implementation: RisImplementation = RisImplementation.BASE,
    strict: bool = True,
    **kw,
```

File: rispy/parser.py
Comment (J535D165) [inline rispy/parser.py:209]: Remove? See comment above.

File: rispy/parser.py
Comment (J535D165) [reply rispy/parser.py:190]: I think it should also be added to the docstring.

**Review by J535D165 — CHANGES_REQUESTED** [review]
This might be a nice feature to have as there is no standard encoding for RIS.

File: tests/test_parser.py
Comment (scott-8) [reply tests/test_parser.py:305]: @shapiromatron still want me to remove this part?

Comment (shapiromatron) [reply tests/test_parser.py:305]: i'd get rid of the variables encoding1 and encoding2 and just write in the load() method calls. then its gtg!

File: README.rst
Comment (shapiromatron) [inline README.rst:31]: too many ps ;)

File: rispy/parser.py
Comment (shapiromatron) [inline rispy/parser.py:302]: add "Consistent with the python standard library, if None is supplied, the default system encoding is used"

**Review by shapiromatron — CHANGES_REQUESTED** [review]
almost there!

File: README.rst
Comment (scott-8) [reply README.rst:31]: the woes of copy pasting

**Review by shapiromatron — APPROVED** [review]

**Pull request merged**
